src/mewbot/tools/lint.py

```diff
@@ -90,13 +93,30 @@ def lint_mypy(self) -> Iterable[Annotation]:
         based on those annotations and resolvable constants.
         """
 
-        args = ["mypy", "--strict", "--exclude", "setup"]
+        args = ["mypy", "--strict", "--explicit-package-bases"]
 
         if not self.is_ci:
             args.append("--pretty")
 
+        # MyPy does not use the stock import engine for doing its analysis,
+        # so we have to give it additional hints about how the namespace package
+        # structure works.
+        # See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules
+        #
+        # There are two steps to this:
+        #  - We set MYPYPATH equivalent to PYTHONPATH
+        os.putenv("MYPYPATH", os.pathsep.join(gather_paths("src")))
```

Comment:
is it worth having a single func that reads "return_os_path(in)"; such as:
```suggestion
        os.putenv("MYPYPATH", return_os_path("src"))
```

Reply:
Adding the rest of this code is A Lot for a suggestion, but I can do to a commit if desired

Reply:
Would that function be anything more than
```
def gather_os_path_str(*folders: str) -> str:
  return os.pathsep.join(gather_paths(*folders))
```

Reply:
Discussed elsewhere, doesn't seem particularly worthwhile where there would only be two callees.